packages/legend-application/src/stores/storage/StorageService.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type PlainObject,
+  SerializationFactory,
+  usingModelSchema,
+} from '@finos/legend-shared';
+import { createModelSchema, list, primitive } from 'serializr';
+import type { GenericLegendApplicationStore } from '../ApplicationStore.js';
+import { LocalStorage } from './LocalStorage.js';
+
+const APPLICATION_SETTINGS_STORAGE_KEY = 'application-settings-storage';
+
+export class StorageItem {
+  key!: string;
+  value!: string;
+
+  constructor(key: string, value: string) {
+    this.key = key;
+    this.value = value;
+  }
+
+  static create(key: string, value: string): StorageItem {
+    return new StorageItem(key, value);
+  }
+}
+
+const storageItemModelSchema = createModelSchema(StorageItem, {
+  key: primitive(),
+  value: primitive(),
+});
+
+export class StorageStore {
+  items: StorageItem[] = [];
+
+  static readonly serialization = new SerializationFactory(
```

Comment:
don't think we need to serialize here at all, `JSON.parse` with `try ... catch` is already good enough, there is no schema to this for now.